src/client/datascience/interactive-common/notebookServerProvider.ts

```diff
@@ -61,7 +62,8 @@ export class NotebookServerProvider implements IJupyterServerProvider {
     ): Promise<INotebookServer | undefined> {
         // When we finally try to create a server, update our flag indicating if we're going to allow UI or not. This
         // allows the server to be attempted without a UI, but a future request can come in and use the same startup
-        this.allowingUI = options.disableUI ? this.allowingUI : true;
+        this.ui.disableUI = options.ui.disableUI;
```

Comment:
Instead of having the event, could you just set the ```this.ui``` to the ```options.ui```? Then it would change when the options changed. Wouldn't need the dispose then.

Reply:
Never mind. This could be changed to not use the event, but you might need the event later.